python/lsst/ts/MTAlignment/alignment_csc.py

```diff
@@ -144,191 +188,359 @@ async def configure(self, config):
+        ----------
+        data : ``cmd_saveJobFile.DataType``
+            Command data.
+        """
         self.assert_enabled()
+        assert self.model is not None
+
         await self.model.save_sa_jobfile(data.file)
 
-    async def correction_loop(self):
+        # TODO: Publish something?
+
+    async def measure_alignment(self, target: str) -> None:
+        """Measure the offset between M1M3 optimum position and the target.
+
+        Parameters
+        ----------
+        target : `str`
+            Target to measure offset with respect to M1M3 optimum position.
+        """
+
+        assert self.model is not None
+
         await self.model.set_telescope_position(
-            self.elevation, self.azimuth, self.camrot
+            telalt=self.elevation,
+            telaz=self.azimuth,
+            camrot=self.camrot,
         )
 
         await self.model.measure_target("M1M3")
-        aligned = False
-        loopcount = 1
-        while not aligned:
-            self.log.info(f"Loop iteration: {loopcount}")
-            await self.model.measure_target("CAM")
-            await self.model.measure_target("M2")
-
-            cam_offset = self.parse_offsets(await self.model.get_target_offset("CAM"))
-            m2_offset = self.parse_offsets(await self.model.get_target_offset("M2"))
-
-            if self.in_tolerance(cam_offset) and self.in_tolerance(m2_offset):
-                break
-            elif loopcount >= self.max_iters:
-                break
-            else:
-                loopcount += 1
 
-    def parse_offsets(self, t2sa_string):
+        if target != "M1M3":
+            await self.model.measure_target(target)
+
+        target_offset = await self.model.get_target_offset(
+            target=target, reference_pointgroup="M1M3"
+        )
+
+        await self.evt_offsetsPublish.set_write(**target_offset)
+
+    async def run_telemetry_loop(self) -> None:
+        """Run telemetry loop.
+
+        The telemetry loop will run while the CSC is in disable or enabled.
         """
-        Takes a string containing spatial coordinates  from T2SA, and returns
-        a dict with the following keys: RefFrame, X, Y, Z, Rx, Ry, Rz, and
-        Timestamp
 
-        Parameters
-        ----------
+        assert self.model is not None
+
+        while self._run_telemetry_loop:
 
-        t2sa_string : `str`
-            the ascii string from T2SA. We expect an ascii string delimited
-            with colons and semicolons, formatted like this:
+            status = await self.model.check_status()
+            if status == "READY":
+                self.laser_status_ready.set()
+            else:
+                self.laser_status_ready.clear()
+
+            # TODO: Publish events with telemetry data.
+
+            await asyncio.sleep(self.heartbeat_interval)
+
+    async def handle_stop_telemetry_loop(self) -> None:
+        """Utility method to handle stopping the telemetry loop and cleaning
```

Comment:
Please consider: "Stop the telemetry loop and clean up".

I'm not convinced this will have any effect on sockets, as long as you actually close them (as you should). I'd be happier if you skipped it until you demonstrated a problem.